interactions-transitions-assignment/part-2-map/us-station-interaction.js

```diff
@@ -22,15 +22,169 @@ function createMap(elementId) {
+
+        circles.exit()
+            .transition().duration(500)
+            .attr("transform", `translate(0,${height})`) // move off screen down
+            .style("opacity", 0)
+            .remove();
+
+        circles.enter().append("circle")
+            .attr("class", "station")
+            .attr("cx", d => projection(d.loc)[0])
+            .attr("cy", d => -50) // start above the screen
+            .attr("r", d => elevScale(d.elevation))
+            .style("opacity", 0)
+            .style("fill", d => stationColorKey(d.stationClass))
+            .merge(circles) 
+            .transition().duration(500)
+            .style("opacity", 0.7)
+            .attr("cx", d => projection(d.loc)[0])
+            .attr("cy", d => projection(d.loc)[1]);
+
+    }
+    function drawStationsCheckbox(stations) {
+        const projection = d3.geoAlbersUsa().fitSize([innerWidth, innerHeight], geoJson);
+    
+        const circles = g.selectAll(".station")
+            .data(stations, d => d.usaf);
+    
+        circles.exit()
+            .transition().duration(500)
+            .attr("r", 0) // shrink to radius 0
+            .style("opacity", 0)
+            .remove();
+    
+        circles.enter().append("circle")
+            .attr("class", "station")
+            .attr("cx", d => projection(d.loc)[0])
+            .attr("cy", d => projection(d.loc)[1])
+            .attr("r", 0) 
+            .style("opacity", 0)
+            .style("fill", d => stationColorKey(d.stationClass))
+            .merge(circles) 
+            .transition().duration(500)
+            .attr("r", d => elevScale(d.elevation)) // grow to appropriate size
+            .style("opacity", 0.7)
+            .attr("cx", d => projection(d.loc)[0])
+            .attr("cy", d => projection(d.loc)[1]);
+    }
+    
+
+    function updateStations() {
+        const selectedClasses = new Set(
+            d3.selectAll("#checkbox-container input:checked").nodes().map(d => +d.value)
+        );
+
+        const minElev = +d3.select("#min-elevation").property("value") || d3.min(nonNullStations, d => d.elevation);
+        const maxElev = +d3.select("#max-elevation").property("value") || d3.max(nonNullStations, d => d.elevation);
+
+        const filteredStations = nonNullStations.filter(d =>
+            selectedClasses.has(d.stationClass) && d.elevation >= minElev && d.elevation <= maxElev
+        );
+
+        drawStationsCheckbox(filteredStations);
+    }
+
+    function handleFilter() {
+        const minElev = +d3.select("#min-elevation").property("value");
+        const maxElev = +d3.select("#max-elevation").property("value");
+
+        if (isNaN(minElev) || isNaN(maxElev)) {
+            alert('Input elevations must be numbers.');
+            return;
+        }
+
+        const filteredStations = nonNullStations.filter(d =>
+            d.elevation >= minElev && d.elevation <= maxElev
+        );
+
+        drawStations(filteredStations);
     }
 }
```

Comment:
it looks like if no value is provided when clicking filter stations, the default is "". You're turning that into a number, which I believe in weird js sorcery will turn the value to false and then to 0. you may want to provide a default value of d3 max/min in order to avoid this.

Reply:
the line +d3.select("#max-elevation").property("value"); is what turns it into a number. You have the alert below it, but I don't think that will ever hit because if there's any invalid number it will be coerced into a number by the "+". If it's not a number I belive it will be false, then 0